### PR TITLE
Python code execute against unit tests strategy

### PR DESCRIPTION
**Pull request by gogo4ds**
Add Python Code execute against unit tests execution strategy
closes https://github.com/SoftUni-Internal/suls-issues/issues/5535

### CODE REVIEW

File: OJS.Workers.Common/Helpers/FileHelpers.cs [hunk 1]
```diff
@@ -161,6 +161,9 @@ public static string ExtractFileFromZip(string pathToArchive, string fileName, s
 
         public static string ProcessModulePath(string path) => path.Replace('\\', '/');
 
+        public static void WriteAllText(string filePath, string text)
+            => File.WriteAllText(filePath, text);
```
Comment (Minkov) [inline OJS.Workers.Common/Helpers/FileHelpers.cs]: Use the service here as proposed in the other PR

File: OJS.Workers.ExecutionStrategies/Python/PythonCodeExecuteAgainstUnitTestsExecutionStrategy.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                var testResult = this.CheckAndGetTestResult(
+                    test,
+                    processExecutionResult,
+                    checker,
+                    message);
+
+                result.Results.Add(testResult);
+            }
+
+            return result;
```
Comment (Minkov) [inline OJS.Workers.ExecutionStrategies/Python/PythonCodeExecuteAgainstUnitTestsExecutionStrategy.cs]: IMO, this looks better:
```
protected override IExecutionResult<TestResult> RunTests(
    string codeSavePath,
    IExecutor executor,
    IChecker checker,
    IExecutionContext<TestsInputModel> executionContext,
    IExecutionResult<TestResult> result)
{
    result.results = executionContext.Input.Tests
        .Select(test => this.DO_YOUR_STUFF_WITH_TEST(codeSavePath, test, executor, checker, executionContext));
}

private THE_TEST_CLASS DO_YOUR_STUFF_WITH_TEST(codeSavePath, test, executor, checker, executionContext) {
    this.WriteTestInCodeFile(executionContext.Code, codeSavePath, test.Input);
    var processExecutionResult = this.Execute(executionContext, executor, codeSavePath, string.Empty);
     / ...

    var testResult = this.CheckAndGetTestResult(
        test,
        processExecutionResult,
        checker,
        message);
    return testResult;
}
```

File: OJS.Workers.ExecutionStrategies/Python/PythonCodeExecuteAgainstUnitTestsExecutionStrategy.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                processExecutionResult.ErrorOutput = string.Empty;
+                processExecutionResult.Type = ProcessExecutionResultType.Success;
+            }
+        }
+
+        private void WriteTestInCodeFile(string code, string codeSavePath, string testContent)
+        {
+            var codeAndTestText = code + Environment.NewLine + testContent;
+
+            FileHelpers.WriteAllText(codeSavePath, codeAndTestText);
```
Comment (Minkov) [inline OJS.Workers.ExecutionStrategies/Python/PythonCodeExecuteAgainstUnitTestsExecutionStrategy.cs]: The service...

File: OJS.Workers.ExecutionStrategies/Python/PythonExecuteAndCheckExecutionStrategy.cs [hunk 4]
```diff
@@ -74,10 +61,30 @@ public class PythonExecuteAndCheckExecutionStrategy : BaseInterpretedCodeExecuti
...
+                    test,
+                    processExecutionResult,
+                    checker,
+                    processExecutionResult.ReceivedOutput);
 
-        private ProcessExecutionResult Execute<TInput>(
+                result.Results.Add(testResult);
+            }
+
+            return result;
```
Comment (Minkov) [inline OJS.Workers.ExecutionStrategies/Python/PythonExecuteAndCheckExecutionStrategy.cs]: Same as described above

File: OJS.Workers.Common/Helpers/FileHelpers.cs (hunk 1 above)
Comment (gogo4ds) [reply OJS.Workers.Common/Helpers/FileHelpers.cs]: The infrastructure does not support dependency injection and service is redundant at this point.

Comment (achobanov) [reply OJS.Workers.Common/Helpers/FileHelpers.cs]: What about factory instantiation?

Comment (gogo4ds) [reply OJS.Workers.Common/Helpers/FileHelpers.cs]: Made a new issue: https://github.com/SoftUni-Internal/interactive-issues/issues/994

File: OJS.Workers.ExecutionStrategies/Python/PythonCodeExecuteAgainstUnitTestsExecutionStrategy.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+                processExecutionResult.ErrorOutput = errorMatch.Groups[1].Value;
+                processExecutionResult.Type = ProcessExecutionResultType.RunTimeError;
+            }
+            else if (failedTestMatch.Success)
+            {
+                message = failedTestMatch.Groups[1].Value;
+            }
+            else if (SuccessTestsRegex.IsMatch(processExecutionResult.ReceivedOutput))
+            {
+                message = "Test Passed!";
```
Comment (Vox94) [inline OJS.Workers.ExecutionStrategies/Python/PythonCodeExecuteAgainstUnitTestsExecutionStrategy.cs]: Consider moving to a global constants file, as this is used in other strategies as well

Comment (gogo4ds) [reply OJS.Workers.ExecutionStrategies/Python/PythonCodeExecuteAgainstUnitTestsExecutionStrategy.cs]: It is moved in the other PR

**Pull request merged**